core/src/main/java/org/elasticsearch/script/ScriptContext.java

```diff
@@ -19,87 +19,35 @@
 
 package org.elasticsearch.script;
 
-import org.elasticsearch.common.Strings;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
 
 /**
- * Context of an operation that uses scripts as part of its execution.
+ * A holder for information about a context in which a script is compiled and run.
  */
-public interface ScriptContext {
-
-    /**
-     * @return the name of the operation
-     */
-    String getKey();
-
-    /**
-     * Standard operations that make use of scripts as part of their execution.
-     * Note that the suggest api is considered part of search for simplicity, as well as the percolate api.
-     */
-    enum Standard implements ScriptContext {
-
-        AGGS("aggs"), SEARCH("search"), UPDATE("update"), INGEST("ingest");
-
-        private final String key;
-
-        Standard(String key) {
-            this.key = key;
-        }
-
-        @Override
-        public String getKey() {
-            return key;
-        }
-
-        @Override
-        public String toString() {
-            return getKey();
-        }
+public final class ScriptContext {
+
+    public static final ScriptContext AGGS = new ScriptContext("aggs");
+    public static final ScriptContext SEARCH = new ScriptContext("search");
+    public static final ScriptContext UPDATE = new ScriptContext("update");
+    public static final ScriptContext INGEST = new ScriptContext("ingest");
+
+    public static final Map<String, ScriptContext> BUILTINS;
```

Comment:
I couldn't find a better name, so we will keep this for now. We can always rename in the future.